feat: Implement Guest Book contract and frontend with `near-sdk-js`

think-in-universe opened the pull request:
JS contract based Guest Book is available for preview at: https://think-in-universe.github.io/guest-book/

The PR includes 

1. `/contract` that is implemented with `near-sdk-js`. Please notice that because `Vector` is not released in the latest [v0.1.1 version](https://www.npmjs.com/package/near-sdk-js), you need to clone [`near-sdk-js` repo](https://github.com/near/near-sdk-js) and put it next to the guest-book folder, then you'll compile the contract successfully. 
2. The frontend is changed slightly to call `call_js_contract` and `view_js_contract` methods on JSVM. The `encodeCall()` should be used very carefully to make the view and call functions work correctly. Default `deposit` should be attached for each call to cover storage cost. 

The intent of the PR is not to merge the JS contract now, but to make it easy to review the required changes to migrate from AS contract to JS contract.

volovyk-s commented on contract/README.md:
```diff
@@ -0,0 +1,16 @@
+
+## Build the contract
+
+First ensure JSVM contract is build and deployed locally, follow [Local Installation](https://github.com/near/near-sdk-js#local-installation). Then run:
```
This is not necessary anymore. Just use `near-sdk-js` package from the NPM.

volovyk-s commented on contract/README.md:
````diff
@@ -0,0 +1,16 @@
+
+First ensure JSVM contract is build and deployed locally, follow [Local Installation](https://github.com/near/near-sdk-js#local-installation). Then run:
+```
+npm i
````
Usually, we are using yarn. For example, all the .lock files are for yarn.

volovyk-s commented on contract/package.json:
```diff
@@ -0,0 +1,20 @@
+  "author": "Near Inc <hello@nearprotocol.com>",
+  "license": "Apache-2.0",
+  "dependencies": {
+    "near-sdk-js": "file:../../near-sdk-js"
```
You can use a package from npm.

volovyk-s commented on src/App.js:
```diff
@@ -9,12 +9,43 @@ import Messages from './components/Messages';
 const BOATLOAD_OF_GAS = Big(3).times(10 ** 13).toFixed();
 
-const App = ({ contract, currentUser, nearConfig, wallet }) => {
+const encodeCall = (contractAccount, methodName, args) => {
```
I believe it will not work with AS contract anymore. We should build some abstraction APIs in the future. Not sure if it is a good idea to merge JS and AS/RS contracts with a common frontend. cc @BenKurrek

volovyk-s commented on src/index.js:
```diff
@@ -31,32 +31,36 @@ async function initContract() {
     {
       // View methods are read-only – they don't modify the state, but usually return some value
-      viewMethods: ['getMessages'],
+      viewMethods: ['view_js_contract'],
```
Yeah, we definitely need to find ad better way. cc @BenKurrek

ailisp replied:
Serhii is right, although it's unlikely we can have a better API in near-api-js before sdk-js 1.0 release. So this is good for this PR

volovyk-s replied:
@ailisp it is good for the presentation, but since it will break the AS example - we should not merge it now. This is my main point.

volovyk-s reviewed (commented):
Do you want to merge this PR into `master`? If so, probably we will need to rename the contract root folder from `contract` to `contract_js` or something like that. The same goes for the `AS` version. Also, we have an issue with the front end. It will now work with AS contract anymore.

ailisp commented:
Code looks good! As Serhii said, you can use near-sdk-js from npm and avoid cloning near-sdk-js repo now.

think-in-universe commented:
@volovyk-s @ailisp I have updated `package.json` with `near-sdk-js` from npm and updated README using near-cli JS commands for deploy and init. 

I didn't update the frontend code. maybe it's a good idea to have a standalone guest-book example for JS contract and frontend. let me know how you think.
